src/Microsoft.NetCore.Analyzers/UnitTests/Security/SslProtocolsAnalyzerTests.cs

```diff
@@ -0,0 +1,548 @@
+        SslProtocols protocols = SslProtocols.Tls11 | SslProtocols.Tls12;
+    }
+}",
+            GetCSharpResultAt(10, 34, SslProtocolsAnalyzer.DeprecatedRule, "Tls11"),
+            GetCSharpResultAt(10, 55, SslProtocolsAnalyzer.HardcodedRule, "Tls12"));
+        }
+
+        [Fact]
+        public void DocSample1_VB_Violation()
+        {
+            VerifyBasic(@"
+Imports System
+Imports System.Security.Authentication
+
+Public Class TestClass
+    Public Sub ExampleMethod()
+        ' CA5395 violation for using Tls11
+        Dim sslProtocols As SslProtocols = SslProtocols.Tls11 Or SslProtocols.Tls12
+    End Sub
+End Class
+",
+            GetBasicResultAt(8, 44, SslProtocolsAnalyzer.DeprecatedRule, "Tls11"),
+            GetBasicResultAt(8, 66, SslProtocolsAnalyzer.HardcodedRule, "Tls12"));
+        }
+
+        [Fact]
+        public void DocSample2_CSharp_Violation()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Security.Authentication;
+
+public class ExampleClass
+{
+    public void ExampleMethod()
+    {
+        // CA5395 violation
+        SslProtocols sslProtocols = (SslProtocols) 768;    // TLS 1.1
+    }
+}",
+            GetCSharpResultAt(10, 37, SslProtocolsAnalyzer.DeprecatedRule, "768"));
+        }
+
+        [Fact]
+        public void DocSample2_VB_Violation()
+        {
+            VerifyBasic(@"
+Imports System
+Imports System.Security.Authentication
+
+Public Class TestClass
+    Public Sub ExampleMethod()
+        ' CA5395 violation
+        Dim sslProtocols As SslProtocols = CType(768, SslProtocols)   ' TLS 1.1
+    End Sub
+End Class
+",
+            GetBasicResultAt(8, 44, SslProtocolsAnalyzer.DeprecatedRule, "768"));
+        }
+
+        [Fact]
+        public void DocSample1_CSharp_Solution()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Security.Authentication;
+
+public class TestClass
+{
+    public void Method()
+    {
+        // In .NET Framework 4.7.1 or later, let the operating system decide what TLS protocol version to use.
+        // See https://docs.microsoft.com/dotnet/framework/network-programming/tls
+        SslProtocols sslProtocols = SslProtocols.None;
+    }
+}");
+        }
+
+        [Fact]
+        public void DocSample1_VB_Solution()
+        {
+            VerifyBasic(@"
+Imports System
+Imports System.Security.Authentication
+
+Public Class TestClass
+    Public Sub ExampleMethod()
+        ' In .NET Framework 4.7.1 or later, let the operating system decide what TLS protocol version to use.
+        ' See https://docs.microsoft.com/dotnet/framework/network-programming/tls
+        Dim sslProtocols As SslProtocols = SslProtocols.None
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void DocSample3_CSharp_Violation()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Security.Authentication;
+
+public class ExampleClass
+{
+    public void ExampleMethod()
+    {
+        // CA5396 violation
```

Comment:
Similar to last one, this file's comments and the analyzers file disagree on the numbering:
Analyzers: 5397 and 5398,
test file: 5395 and 5396